### PR TITLE
CORE-6776: create combined worker installer scripts 

### PR DESCRIPTION
**Pull request by ronanbrowne**
As the combined worker is a runnable and not a consumable library we want people to install it via a script / downland, rather than consume it via Maven Central. 

Adding gradle logic which will create a new archive zip produced by the combined worker module, this archive contains the following.

- a shell script
- a PowerShell script which will place the combined work
- the actual combined worker jar.

These scripts add the combined worker to a users ~/corda directory and also generate a scrip which will start the combined worker, Further parameters can be passed to these scripts as needed e.g. combined worker DB details etc.

This is inline with how the [corda-cli scripts](https://github.com/corda/corda-runtime-os/tree/release/os/5.0/tools/plugins/installScripts) currently work.

Further to this line ending updated specifically for *.sh to address errors some users were experiencing with line endigns.

A further piece of work will add a script that will download this from S3 which we will provide a direct link to allowing a user to run the following ....curl "https://xxxxxxxxxx/get-corda-combined-worker.sh" | sudo bash , this script will in turn download this new zip archive added in this PR, inflate it and generate a start-combined-worker,sh script

### CODE REVIEW

File: applications/workers/release/combined-worker/installScripts/installCombinedWorker.ps1 [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+$combinedWorkerCommand = "$ENV:JAVA_HOME\bin\java -jar corda-combined-worker.jar %*"
+New-Item "$combinedWorkerDir\corda-combined-worker.cmd" -ItemType File -Value $combinedWorkerCommand
+
+if($addToPath) {
+    Write-Output "Adding combined-worker to path"
+
+    Get-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name path
+    $old = (Get-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name path).path
+    $new  =  "$old;$combinedWorkerDir"
+    Set-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name path -Value $new
```
Comment (yarivus) [inline applications/workers/release/combined-worker/installScripts/installCombinedWorker.ps1:21]: Something doesn't make sense here ... it seems you're calling `Get-ItemProperty` twice redundantly.
Also, I think it  would be beneficial to save the string:
```
'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment'
```
in a temp var.
So my suggestion for the new code:
```
$regPath = 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment'
$oldRegPath = (Get-ItemProperty -Path $regPath -Name path).path
$newRegPath = "$oldRegPath;$combinedWorkerDir"
Set-ItemProperty -Path $regPath -Name path -Value $newRegPath
```

**Review by yarivus — CHANGES_REQUESTED** [review]
see my comments @ronanbrowne

**Comment by driessamyn** [comment]
No objections with the PR, but out of interest, why are we doing this?
The `combined-worker` is supposed to be an internal configuration for R3 Corda Developers for debugging and Corda. My understanding was that it will also, temporarily, be packaged up with the DevEx tooling for DP2, but why does it need an installer?

**Comment by ronanbrowne** [comment]
> No objections with the PR, but out of interest, why are we doing this? The `combined-worker` is supposed to be an internal configuration for R3 Corda Developers for debugging and Corda. My understanding was that it will also, temporarily, be packaged up with the DevEx tooling for DP2, but why does it need an installer?

@driessamyn sure thanks for feedback i wanted to start the conversation around the consumption of the combined worker. my understanding is as this is a runnable, it's not something we were going to publish to maven central, in the same way we are not putting the cli host there it will be installed via a script.

There's a Q here if we are happy with people consuming the combined worker directly as a dependency in peoples cordApps ?

e.g.

    combinedWorker "net.corda:corda-combined-worker:$combinedWorkerVersion"
    
But if this is the preferred approach for DP2, we can descope this.

File: applications/workers/release/combined-worker/installScripts/installCombinedWorker.ps1 (hunk 1 above)
Comment (ronanbrowne) [reply applications/workers/release/combined-worker/installScripts/installCombinedWorker.ps1:21]: Thanks, will hold off on updating based on outcome of below conversations

**Comment by driessamyn** [comment]
@ronanbrowne , I thought the combined worker would only be consumed through the CSDE tooling, but @mattbradburyr3 should be able to confirm.

Again, I don't have an objection against this, but wonder if it's un-necessary work, and possibly un-necessary support follow-up.

**Comment by ronanbrowne** [comment]
> @ronanbrowne , I thought the combined worker would only be consumed through the CSDE tooling, but @mattbradburyr3 should be able to confirm.
> 
> Again, I don't have an objection against this, but wonder if it's un-necessary work, and possibly un-necessary support follow-up.

Currently yes that's correct , I have a meeting later with Matthew will discuss then , If this is too big a change of interaction for DP2 it's something we can discuss for Beta - moving to draft until that discussion happens

**Comment by ronanbrowne** [comment]
FYI Closing this for DP2 after conversations with dev ex, will return to this for BETA if  required . CC @driessamyn  / @mattbradburyr3

**Pull request closed without merging**